presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java

```diff
@@ -514,29 +514,37 @@ public boolean mkdirs(Path f, FsPermission permission)
             key += PATH_SEPARATOR;
         }
 
-        ListObjectsRequest request = new ListObjectsRequest()
+        ListObjectsV2Request request = new ListObjectsV2Request()
                 .withBucketName(getBucketName(uri))
                 .withPrefix(key)
                 .withDelimiter(PATH_SEPARATOR)
                 .withRequesterPays(requesterPaysEnabled);
 
         STATS.newListObjectsCall();
-        Iterator<ObjectListing> listings = new AbstractSequentialIterator<ObjectListing>(s3.listObjects(request))
+        Iterator<ListObjectsV2Result> listings = new AbstractSequentialIterator<ListObjectsV2Result>(s3.listObjectsV2(request))
         {
             @Override
-            protected ObjectListing computeNext(ObjectListing previous)
+            protected ListObjectsV2Result computeNext(ListObjectsV2Result previous)
             {
                 if (!previous.isTruncated()) {
                     return null;
                 }
-                return s3.listNextBatchOfObjects(previous);
+
+                final ListObjectsV2Request nextRequest = new ListObjectsV2Request()
```

Comment:
you can use previous `request` and just set continuation token, see: https://docs.aws.amazon.com/AmazonS3/latest/dev/ListingObjectKeysUsingJava.html

Reply:
Sorry for the late reply, I will change code and make a basic test today.